ruby/engine/embedded_help.rb

```diff
@@ -637,45 +639,61 @@ def self.glob(pattern, *args, **options)
       pattern_array = pattern
     end
 
-    #puts "Dir.glob pattern = #{pattern}, pattern_array = #{pattern_array}, args = #{args}, options = #{options}"
+    pattern_has_embedded = pattern_array.any? {|p| p.to_s.chars.first == ':'}
+    base_has_embedded = (!base.nil? && base.to_s.chars.first == ':')
+    if !pattern_has_embedded && !base_has_embedded
+      # puts "Original glob"
+      return self.original_glob(pattern, flags: flags, base: base, sort: sort)
+    end
+
+    absolute_base = if base.nil?
+                      nil
+                    else
+                      OpenStudio.get_absolute_path(base)
+                    end
+    if debug
+      puts "pattern_array=#{pattern_array}"
+      puts "base=#{base}"
+      puts "flags=#{flags}"
+      puts "pattern_has_embedded=#{pattern_has_embedded}"
+      puts "base_has_embedded=#{base_has_embedded}"
+      puts "absolute_base=#{absolute_base}"
+    end
+
+    # DLM: seems like this is needed for embedded paths, possibly due to leading ':' character?
     override_args_extglob = false
 
     result = []
     pattern_array.each do |pattern|
 
-      if pattern.to_s.chars.first == ':'
-
-        # DLM: seems like this is needed for embedded paths, possibly due to leading ':' character?
-        override_args_extglob = true
-
-        #puts "searching embedded files for #{pattern}"
-        absolute_pattern = OpenStudio.get_absolute_path(pattern)
-        #puts "absolute_pattern #{absolute_pattern}"
-
-        EmbeddedScripting::fileNames.each do |name|
-          absolute_path = OpenStudio.get_absolute_path(name)
+      absolute_pattern = if pattern.to_s.chars.first == ':'
+                           OpenStudio.get_absolute_path(pattern)
+                         else
+                           pattern
+                         end
+      if debug
+        puts "searching embedded files for #{pattern}"
+        puts "absolute_pattern #{absolute_pattern}"
+      end
 
-          if override_args_extglob
-            if File.fnmatch( absolute_pattern, absolute_path, File::FNM_EXTGLOB )
-              #puts "#{absolute_path} is a match!"
-              result << absolute_path
-            end
-          else
-            if File.fnmatch( absolute_pattern, absolute_path, *args, **options )
-              #puts "#{absolute_path} is a match!"
-              result << absolute_path
-            end
+      EmbeddedScripting::fileNames.each do |name|
+        absolute_path = OpenStudio.get_absolute_path(name)
+        if base_has_embedded
+          next unless absolute_path.start_with?(absolute_base)
+          if debug
+            puts "name=#{name}, absolute_path=#{absolute_path}"
+            puts "absolute_path.start_with?(absolute_base)=#{absolute_path.start_with?(absolute_base)}"
```

Comment:
loop on the embedded fileNames (which all start with ':/'), skip if they don't match the base